### PR TITLE
use Guard from Polyfill 

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: src/Adapter/MSTest.TestAdapter/ObjectModel/TestMethod.cs [hunk 1]
```diff
@@ -43,10 +43,7 @@ public TestMethod(string name, string fullClassName, string assemblyName, bool i
         string fullClassName, string assemblyName, bool isAsync, string? displayName,
         TestIdGenerationStrategy testIdGenerationStrategy)
     {
-        if (StringEx.IsNullOrEmpty(assemblyName))
```
Comment (SimonCropp) [inline src/Adapter/MSTest.TestAdapter/ObjectModel/TestMethod.cs:46]: i assume whitespace is also invalid?

File: src/Adapter/MSTest.TestAdapter/GlobalUsings.cs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+global using Polyfills;
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/GlobalUsings.cs]: Could you use global usings from the csproj please?

File: src/Adapter/MSTest.TestAdapter/ObjectModel/TestMethod.cs (hunk 1 above)
Comment (Evangelink) [reply src/Adapter/MSTest.TestAdapter/ObjectModel/TestMethod.cs:46]: I would say so!

**Review by Evangelink — DISMISSED** [review]
Can you please define global using in the csproj?

**Review by Evangelink — APPROVED** [review]

**Comment by Evangelink** [comment]
I have triggered sync of the package, I'll update the PR once it's done

**Comment by Evangelink** [comment]
Alright so tests are failing because we force the locally built version of the platform that contains a breaking change (ArgumentGuard is gone). So I'll force merge and skip tests and do the roundtrip insertion.

**Pull request merged**